src/ComposerRequireChecker/FileLocator/LocateComposerPackageDirectDependenciesSourceFiles.php

```diff
@@ -13,23 +16,46 @@ public function __invoke(string $composerJsonPath): Generator
 
         $composerJson = json_decode(file_get_contents($composerJsonPath), true);
         $configVendorDir = $composerJson['config']['vendor-dir'] ?? 'vendor';
-        $vendorDirs = array_values(
-            array_map(
-                function (string $vendorName) use ($packageDir, $configVendorDir) {
-                    return $packageDir . '/' . $configVendorDir . '/' . $vendorName;
-                },
-                array_keys($composerJson['require'] ?? [])
-            )
-        );
-
-        foreach ($vendorDirs as $vendorDir) {
-            if (!file_exists($vendorDir . '/composer.json')) {
+        $vendorDirs = [];
+        foreach ($composerJson['require'] ?? [] as $vendorName => $vendorRequiredVersion) {
+            $vendorDirs[$vendorName] = $packageDir . '/' . $configVendorDir . '/' . $vendorName;
+        };
+
+        if (empty($vendorDirs)) {
```

Comment:
Is this block necessary?

Also: please don't use `empty()`, as it is too lax

Reply:
Well if this is empty then we can save the effort of loading `installed.json`, so not necessary but a minor improvement.

I don't feel strongly either way, happy to remove it or change to `count() === 0`?

Reply:
I'd get rid of it completely: this is an extremely edge case scenario, and an optimisation is not worth the additional execution path

Reply:
Done